src/csharp/Grpc.Core.Api/Metadata.cs

```diff
@@ -83,7 +83,7 @@ public Entry Get(string key)
         {
             for (int i = entries.Count - 1; i >= 0; i--)
             {
-                if (entries[i].Key == key)
+                if (string.Equals(entries[i].Key, key, StringComparison.OrdinalIgnoreCase))
```

Comment:
since NormalizeKey()  uses `key.ToLowerInvariant();`, we'd need to use StringComparison.InvariantCultureIgnoreCase here.

Reply:
String conversion isnt the same as string comparison. If this were using different encodings I would agree but its not and these do not have to "match" in that way. 

Also there isnt a .ToLowerOrdinal() function so .ToLowerInvariant() is the best available choice.  OrdinalIgnoreCase and InvariantCultureIgnoreCase both eschew culture specifics, but Ordinal is preferred for [several reasons](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings). 

And your commit also uses OrdinalIgnoreCase (albeit in an odd and hard to follow manner), so why not resolve or remove this comment?